### PR TITLE
[ARCTIC-351][OPTIMIZE]:Supports external system registration and management of Optimizer

### LINKED ISSUE #351: [Improvement][OPTIMIZE]:Supports external system registration and management of Optimizer
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?page=4&q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

now external system can't  registration and management of Optimizer

### How should we improve?

1.Provides interfaces for registering and unregistering Optimizers
2.Provides an interface to get the optimize Group

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hameizi**
## Why are the changes needed?
resolve #351 

## Brief change log

register optimizer when there is not optimizerId in start optimizer config.

## How was this patch tested?
start optimizer with -a {ams-url} -qn {optimizerGroupName} -p {executor-parallel} -m {executor-memory-size}

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

**Review by Aireed — APPROVED** [review]
LGTM

File: optimizer/src/main/java/com/netease/arctic/optimizer/local/LocalOptimizer.java [hunk 1]
```diff
@@ -150,6 +151,10 @@ public void stop() throws IOException {
 
   public static void main(String[] args) throws CmdLineException {
     OptimizerConfig optimizerConfig = new OptimizerConfig(args);
+    if (optimizerConfig.getOptimizerId() == null || optimizerConfig.getOptimizerId().isEmpty() ||
```
Comment (wangtaohz) [inline optimizer/src/main/java/com/netease/arctic/optimizer/local/LocalOptimizer.java:154]: I think not only external optimizer but also "internal" optimizer can `register` to ams, and then the start script of external/internal optimizer could be unified.

Comment (wangtaohz) [reply optimizer/src/main/java/com/netease/arctic/optimizer/local/LocalOptimizer.java:154]: It confusing that the start script of internal optimizer use `--queue-id`, but external optimizer use `--queue-name`

Comment (hameizi) [reply optimizer/src/main/java/com/netease/arctic/optimizer/local/LocalOptimizer.java:154]: > It confusing that the start script of internal optimizer use `--queue-id`, but external optimizer use `--queue-name`

the start script of `internal` optimize is generate with ams , so the `queue-id` is easy to get. but external optimizer is started by user who get `queue-id` diffcult

Comment (wangtaohz) [reply optimizer/src/main/java/com/netease/arctic/optimizer/local/LocalOptimizer.java:154]: what if user starts external optimizer but set optimizerId?

Comment (hameizi) [reply optimizer/src/main/java/com/netease/arctic/optimizer/local/LocalOptimizer.java:154]: > what if user starts external optimizer but set optimizerId?

it also work well.

**Pull request merged**